src/abstractdataframe/where.jl

```diff
@@ -0,0 +1,203 @@
+# where allows a transformation specification without a target column name or a column
+
+_process_where_pair(i::Int, a::ColumnIndex) = a => Symbol(:x, i)
+_process_where_pair(i::Int, @nospecialize(a::Pair{<:Any, <:Base.Callable})) =
+    first(a) => last(a) => Symbol(:x, i)
+_process_where_pair(i::Int, a) =
+    throw(ArgumentError("Condition specifier $a is not supported by `where`"))
+
+function _get_where_conditions(df::Union{AbstractDataFrame, GroupedDataFrame},
+                               @nospecialize(args), dropmissing::Bool)
+    conditions = Any[_process_where_pair(i, a) for (i, a) in enumerate(args)]
+
+    isempty(conditions) && throw(ArgumentError("At least one condition must be passed"))
+
+    if df isa AbstractDataFrame
+        df_conditions = select(df, conditions..., copycols=!(df isa DataFrame))
+    else
+        df_conditions = select(df, conditions...,
+                               copycols=!(parent(df) isa DataFrame), keepkeys=false)
+    end
+    test_type = dropmissing ? Union{Missing, Bool} : Bool
+    for col in eachcol(df_conditions)
+        if !(eltype(col) <: test_type)
+            throw(ArgumentError("each transformation must return a vector whose " *
+                                "eltype is subtype of $test_type"))
+        end
+    end
+    @assert ncol(df_conditions) == length(conditions)
+    if ncol(df) == 1
+        return .===(df_conditions[!, 1], true)
```

Comment:
We should use `isequal` instead of `===`. It results in a significant compilation speedup for interactive use.

Reply:
Let us discuss it as it was what I originally preferred, but @nalimilan wanted to switch to `===`.

Reply:
That's surprising. In general `===` should be faster as it's very simple (no dispatch). Can you post a benchmark?

Reply:
maybe it is because it is broadcasted and some bad things (invalidations) happen when this mechanics is triggered?

Reply:
Hmm... yes I think I was mistaken. I can't replicate it now.